parlai/core/worlds.py

```diff
@@ -42,22 +42,29 @@
 """
 
 import copy
+import queue
 import random
 import time
 
+from abc import ABC, abstractmethod
+from enum import auto, Enum
 from functools import lru_cache
-from typing import List, Dict, Any, Union
+from typing import List, Dict, Any, Union, Tuple, Optional
 
 try:
-    from torch.multiprocessing import Process, Value, Condition, Semaphore
+    from torch.multiprocessing import Process, Value, Semaphore, Queue
+    import torch
+    import torch.multiprocessing as mp
 except ImportError:
-    from multiprocessing import Process, Value, Semaphore, Condition  # noqa: F401
+    from multiprocessing import Process, Value, Semaphore, Queue  # noqa: F401
 
-from parlai.core.agents import create_agents_from_shared
+from parlai.core.agents import create_agents_from_shared, Agent
 from parlai.core.loader import load_task_module, load_world_module
+from parlai.core.message import Message
 from parlai.core.metrics import aggregate_metrics
 from parlai.core.opt import Opt
 from parlai.core.teachers import create_task_agent_from_taskname
+from parlai.core.torch_agent import Batch
```

Comment:
Hmm I think we need to yank out batch into a new file to prevent torch from being needed for mturk

Reply:
ok, i am moving to `parlai/utils/batch.py`